File: PersonalTimeManager.Server/Models/Task.cs
using Google.Cloud.Firestore;

[FirestoreData]
public class TaskEntity
{
    [FirestoreProperty]
    public string Id { get; set; } = Guid.NewGuid().ToString();

    [FirestoreProperty]
    public string? UserId { get; set; }

    [FirestoreProperty]
    public string Name { get; set; }

    [FirestoreProperty]
    public string Description { get; set; }

    [FirestoreProperty]
    public string Type { get; set; }

    [FirestoreProperty]
    public string? StartTime { get; set; }

    [FirestoreProperty]
    public string EndTime { get; set; }

    [FirestoreProperty]
    public List<string> Dates { get; set; }
}

File: personaltimemanager.client/src/app/models/Task.ts
export interface Task {
  id: string;
  userId: string;
  name: string;
  description: string;
  type: 'ongoing' | 'one-time' | '';
  startTime: string;
  endTime: string;
  dates: string[];
  isCompleted: boolean;
}


export const DEFAULT_TASK: Task = {
  id: '',
  userId: '',
  name: '',
  description: '',
  type: '',
  startTime: '',
  endTime: '',
  dates: [],
  isCompleted: false
}
